src/components/missions/MissionsItem.js

```diff
@@ -1,20 +1,38 @@
 import React from 'react';
+import { useDispatch } from 'react-redux';
 import { PropTypes } from 'prop-types';
 import Badge from 'react-bootstrap/Badge';
 import Button from 'react-bootstrap/Button';
+import { joinMission, leaveMission } from '../../redux/missions/missions';
 
-const MissionsItem = ({ name, description }) => (
-  <tr>
-    <th>{name}</th>
-    <td>{description}</td>
-    <td><Badge bg="secondary">Not A Member</Badge></td>
-    <td colSpan={2}><Button variant="secondary">Join Mission</Button></td>
-  </tr>
-);
+const MissionsItem = ({
+  id, name, description, reserved,
+}) => {
+  const dispatch = useDispatch();
+
+  const handleJoinMission = () => {
+    dispatch(joinMission(id));
+  };
+
+  const handleLeaveMission = () => {
+    dispatch(leaveMission(id));
+  };
+
+  return (
+    <tr>
+      <th>{name}</th>
+      <td>{description}</td>
+      <td>{reserved ? (<Badge bg="info">Active Member</Badge>) : (<Badge bg="secondary">Not A Member</Badge>)}</td>
+      <td colSpan={2}>{reserved ? (<Button onClick={handleLeaveMission} variant="outline-danger">Leave Mission</Button>) : (<Button onClick={handleJoinMission} variant="outline-secondary">Join Mission</Button>)}</td>
+    </tr>
+  );
+};
```

Comment:
Great job with implementing what happens when a user clicks the "Join Mission" button; action dispatched to update the store.  The state isn't mutated.